packages/neo-one-client/src/user/keystore/LedgerHandler.ts

```diff
@@ -0,0 +1,148 @@
+import { crypto } from '@neo-one/client-core';
+import { HWLedger, LedgerTransport } from '@neo-one/client-switch';
+
+export interface LedgerResponse {
+  readonly message: Buffer;
+  readonly status: Buffer;
+}
+
+export interface LedgerQueueRequest {
+  readonly params: string;
+  readonly msg: string;
+  readonly statusList: ReadonlyArray<number>;
+  readonly resolve: (value: LedgerResponse) => void;
+  // tslint:disable-next-line:no-any
+  readonly reject: (error: Error) => void;
+}
+const VALID_STATUS = 0x9000;
+// const MSG_TOO_BIG = 0x6d08;
+// const APP_CLOSED = 0x6e00;
+// const TX_DENIED = 0x6985;
+// const TX_PARSE_ERR = 0x6d07;
+
+const BIP44 = (accountIn = 0) => {
+  const account = accountIn.toString(16);
+
+  return `8000002C800003788000000000000000${'0'.repeat(8 - account.length)}${account}`;
+};
+
+const getParams = (params: string) => {
+  if (params.length !== 8) {
+    throw new Error(`params requires 4 bytes`);
+  }
+  const paramsMatched = params.match(/.{1,2}/g);
+  if (paramsMatched === null) {
+    throw new Error(`send params got matched to null`);
+  }
+
+  return paramsMatched.map((val) => parseInt(val, 16));
+};
+
+export class LedgerHandler {
+  public static readonly byteLimit = LedgerTransport.byteLimit;
+  public static readonly type = LedgerTransport.type;
+
+  public static readonly init = async () => {
+    const isSupported = await LedgerTransport.isSupported();
+    if (!isSupported) {
+      throw new Error(`Ledger not supported by your machine`);
```

Comment:
Use the custom error codes so client api users can catch and identify the errors